### DISCUSSION TITLE
Set dark and light theme

### DISCUSSION THREAD

**Discussion by VideoCarp** [emilk/egui#1627] [Q&A] [answered]
How can I toggle dark theme similarly to [the egui demo](https://egui.rs) and in the repo's README.
I've looked at a previous discussion, but it didn't help. I've looked at the docs but I couldn't understand them very well.

How do I change the theme to dark mode? I'm not asking about how to make a toggle button. I am asking how I can toggle the dark mode.

**Comment by willser** [comment, marked as answer]
````rust
use eframe::egui::{Style, Visuals};

let native_options = eframe::NativeOptions::default();
eframe::run_native(
    "test",
    native_options,
    Box::new(|creation_context| {
        let style = Style {
            visuals: Visuals::dark(),
            ..Style::default()
        };
        creation_context.egui_ctx.set_style(style);
        Box::new(<ur struct>)
    }),
);
````

Or

````rust
ui.style_mut().visuals = Visuals::dark();
````

**Comment by floppyhammer** [comment]
If you want to toggle theme globally
```rust
// ctx: egui::Context
ctx.set_visuals(Visuals::dark());
```

**Comment by landaire** [comment]
The previous answers did not work for me. As of Nov 2024 this variation of willser's answer works:

```rust
eframe::run_native(
    "test",
    native_options,
    Box::new(|creation_context| {
        creation_context.egui_ctx.set_theme(egui::Theme::Dark);

        Box::new(<ur struct>)
    }),
);

```

**Comment by iFrogHop2Worlds** [comment]
ui.checkbox(&mut self.dark_mode_enabled, "Dark Mode").changed().then(|| {
                           if self.dark_mode_enabled {
                               ctx.set_theme(egui::Theme::Dark);
                           } else {
                               ctx.set_theme(egui::Theme::Light);
                           }
                       });
